src/Npgsql/NpgsqlConnection.cs

```diff
@@ -388,13 +401,13 @@ public override string ConnectionString
         /// Backend server host name.
         /// </summary>
         [Browsable(true)]
-        public string? Host => Settings.Host;
+        public string? Host => Connector?.Host;
 
         /// <summary>
         /// Backend server port.
         /// </summary>
         [Browsable(true)]
-        public int Port => Settings.Port;
+        public int? Port => Connector?.Port;
```

Comment:
We should probably  leave this non-nullable and have it default to 0, as that's not a valid port. That would be one less breaking change, and APIs are generally a bit more usable when Nullable<T> can be avoided (i.e. there's a natural "not set" value for T).